src/Model/GServer.php

```diff
@@ -481,6 +474,14 @@ public static function setFailureByUrl(string $url)
 			return;
 		}
 
+		if (Network::isUrlBlocked($url)) {
+			Logger::info('Server domain is blocked', ['url' => $url]);
+			return;
+		} elseif (Network::isUrlBlocked($nurl)) {
+			Logger::info('Server domain is blocked', ['nurl' => $nurl]);
+			return;
+		}
```

Comment:
The reason why I asked you to move these conditions later is that this is a method to update a potentially existing server record. If the record was created earlier (and not purged adding a new domain pattern to the block list) then we do want to keep it accurate, even if it is currently blocked.

On the other hand, we absolutely do not want to create a new record if the domain is blocked. So we need to allow the update case even if the domain is blocked.

Reply:
We don't need this check since we check for blocked domains in the parent functions. After I deleted the server entries from my system several days ago, the current code worked perfectly and hadn't created any new of these entries.

Reply:
Should we revert then?

Reply:
Here it didn't work perfectly. My `gserver` table was still flooded with +1.7 million records non-stop. Only when I added the code, it was finally solved. Okay, I can raise the log level to see if these messages there are showing up.

Reply:
Here is another report of being flooded by these trolls (without this PR): https://fika.grin.hu/display/9b3ec022-4463-d42e-0a21-6b3257115451

Reply:
Yes, he is on 2022.10. We improved the server block checks with 2023.01.